network/modules/UDP/socket_wrapper.py

```diff
@@ -0,0 +1,138 @@
+import socket
+import struct 
+
+class UdpSocket:
+    """
+    Wrapper for Python's socket module.
+    """
+
+    def __init__(self, socket_instance: socket.socket = None) -> None:
+        """
+        Parameters
+        ----------
+        instance: socket.socket
+            For initializing Socket with an existing socket object.
+        """
+        if socket_instance is None:
+            self.__socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        else:
+            self.__socket = socket_instance
+
+    def send_to(self, data: bytes, host: str = "", port: int = 5000) -> bool:
+        """
+        Sends data to specified address
+
+
+        Parameters
+        ----------
+        data: bytes
+        host: str (default "")
+            Empty string is interpreted as '0.0.0.0' (IPv4) or '::' (IPv6), which is an open address
+        port: int (default 5000)
+            The host, combined with the port, will form the address as a tuple
+
+
+        Returns
+        -------
+        bool: if data was transferred successfully
+
+        """
+        address = (host, port)
+
+        data_sent = 0
+        data_size = len(data)
+
+        while data_sent < data_size:
+
+            chunk = data[data_sent:data_sent+4096]
+            packed_data = struct.pack(f'!{len(chunk)}s', chunk)
```

Comment:
Sorry, my bad. There's no need to pack our data using struct, I kind of overthinked for a moment. This is only necessary when you are say sending an array of integers, but we are sending raw bytes to be interpreted by the image_decode module later anyways.